clang/include/clang/Basic/Cuda.h

```diff
@@ -128,12 +128,13 @@ enum class CudaArch {
   GFX12_GENERIC,
   GFX1200,
   GFX1201,
+  AMDGCNSPIRV,
   Generic, // A processor model named 'generic' if the target backend defines a
            // public one.
   LAST,
 
   CudaDefault = CudaArch::SM_52,
-  HIPDefault = CudaArch::GFX906,
+  HIPDefault = CudaArch::AMDGCNSPIRV,
```

Comment:
Why was the default changed here? The default here is just what HIP compilation gives without an explicit arch. Is this intentional?

Reply:
Yes, the intent is for HIP go to to SPIRV for this case, since that'll maximise "usability" of the compiled binary, for lack of a better word.

Reply:
Yeah, makes sense. But doesn't the SPIR-V toolchain require extra tools?

Reply:
In the long run it should not (the BE is almost baked, for example); in the interim, it will need the translator, and I (think I) do see what you are saying, that it might be onerous to have HIPAMD depend on that - I can unflip the default pending the BE coming along. The only cost would be slightly more dubious ergonomics, but those are pre-existing.

Reply:
Did this patch pass our internal PSDB?

I would recommend deferring changing the default offload arch to future while checkin the other changes first.

I doubt the downstream build is ready to have llvm-link and llvm-spirv. Some HIP apps do a test compilation of HIP program with default offload arch and this will break them.

Reply:
Behavior change should definitely be done separately

Reply:
Done.